source/atomic_ops.cpp

```diff
@@ -63,7 +62,56 @@ bool atomic_cas<uint32_t>(uint32_t *ptr, uint32_t *expectedCurrentValue, uint32_
 
     return !__STREXW(desiredValue, ptr);
 }
-#endif /* #if (__CORTEX_M >= 0x03) */
+
+uint32_t atomic_incr(uint32_t *valuePtr, uint32_t delta)
+{
+    uint32_t newValue;
+    do {
+        newValue = __LDREXW(valuePtr) + delta;
```

Comment:
As far as abstraction of ```LDREX``` and ```STREX``` go, I still don't understand why the abstractions need to be unified to a single (complex) API. Can you provide an example that demonstrates why ```atomic_cas``` must be the core of ```atomic_incr/atomic_decr``` instead of a parallel API (when specialized).